lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -647,6 +647,46 @@ class BroadcastOpConversionPattern
   }
 };
 
+class SubtractOpConversionPattern
+    : public OpConversionPattern<ttir::SubtractOp> {
+  using OpConversionPattern<ttir::SubtractOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(ttir::SubtractOp srcOp, ttir::SubtractOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    RankedTensorType lhsType =
+        mlir::cast<RankedTensorType>(adaptor.getInputs().front().getType());
+    RankedTensorType rhsType =
+        mlir::cast<RankedTensorType>(adaptor.getInputs().back().getType());
+
+    if (lhsType.getShape() == rhsType.getShape()) {
+      rewriter.replaceOpWithNewOp<ttnn::SubtractOp>(
+          srcOp, adaptor.getInputs().front(), adaptor.getInputs().back(),
+          adaptor.getOutputs().front());
+
+      // Broadcast for rhs operand require the operation to be commutative to
+      // allow switching the order of operands. To allow this conversion, the
+      // following conversion is applied to SubtractOp: subtractOp(lhs,rhs) ->
+      // addOp(lhs, negOp(rhs))
+
+    } else {
+      Value device = getOrInsertDevice(rewriter, srcOp);
+      tensor::EmptyOp negEmptyOp = rewriter.create<tensor::EmptyOp>(
+          srcOp.getLoc(), this->getTypeConverter()->convertType(rhsType),
+          device);
+      ttnn::NegOp negOp = rewriter.create<ttnn::NegOp>(
+          srcOp.getLoc(), adaptor.getInputs().back(), negEmptyOp);
+
+      rewriter.replaceOpWithNewOp<ttnn::AddOp>(
+          srcOp, adaptor.getInputs().front(), negOp.getResults().front(),
+          adaptor.getOutputs().front());
```

Comment:
Yeah, this is another example of what we've talked about. @uazizTT to give you more context, one of the topics of discussion in F2F meetings this week was how to handle metal issues that we hit on our side. We've agreed that we need to make a mindset shift where we first talk with metal team and see if they can fix it on their end, before making workarounds on our side. So this is the desired pattern of handling issues like this one:
1. Open issue in metal repo and ask for estimate on how long it will take to fix.
2. If they agree to fix it in a reasonable time, then we just wait. If not, then ask how feasible it is to fix, i.e. how much work they think it would take so we can see if we want to help with a fix.
3. If we are okay with fixing it ourselves and making a contribution then we do that. If not, then we figure out what is the best workaround to make on our end.
4. We open issue on our end to revert the workaround when the fix on metal side lands.

@uazizTT Can you please open issues on metal repo (from step 1) and on our repo (from step 4)? If you don't get any response from metal team and need to escalate, you can tag @tt-mpantic to assist.

Reply:
Btw @uazizTT, another thing that we've discussed in F2F meetings was how to improve our PR process. From now on it will be required to always get a sign off from at least one code owner of each code component that you've changed. This will be automated by github rules and we also have to update the codeowners file, but until then, could you please ping the codeowners who didn't approve before you merge just to check if they are okay with that, especially if there are some unresolved comments that they've left. This week many of us were too busy with F2F meetings to review PRs so I understand that you felt blocked, don't worry about it nobody will take it amiss, also this is a fresh thing that we've just discussed and old rules were more relaxed 🙂 The main point is that we are not in a rush to make changes asap and we should invest more time to make sure that we are making the best changes we can, so even if you have to wait a few days to iterate and get approval - in most cases it will be fine.